InscryptionAPI/Card/CardExtensions.cs

```diff
@@ -296,61 +367,175 @@ public static CardInfo SetRare(this CardInfo info)
+        info.evolveParams = new()
+        {
+            evolution = CardLoader.Clone(evolveCard)
+        };
+
+        if (mods != null)
+        {
+            info.evolveParams.evolution = CardLoader.Clone(info.evolveParams.evolution);
+            (info.evolveParams.evolution.mods ??= new ()).AddRange(mods);
+        }
+
+        info.evolveParams.turnsToEvolve = numberOfTurns;
+
+        return info;
+    }
+
+    /// <summary>
+    /// Sets the evolve parameters of the card. These parameters are used to make the Evolve ability function correctly.
+    /// This function uses delayed loading to attach the evolution to the card, so if the evolve card doesn't exist yet, this function will still work.
+    /// </summary>
+    /// <param name="info">Card to access</param>
+    /// <param name="evolveInto">The name of card that will be generated after the set number of turns.</param>
+    /// <param name="numberOfTurns">The number of turns before the card evolves</param>
+    /// <param name="mods">A set of card mods to be applied to the evolved card</param>
+    /// <returns>The same card info so a chain can continue</returns>
+    public static CardInfo SetEvolve(this CardInfo info, string evolveInto, int numberOfTurns, IEnumerable<CardModificationInfo> mods = null)
+    {
+        CardInfo evolution = CardManager.AllCardsCopy.CardByName(evolveInto);
+
+        if (evolution == null) // Try delayed loading
+        {
+            CardManager.ModifyCardList += delegate(List<CardInfo> cards)
+            {
+                CardInfo target = cards.CardByName(info.name);
+                CardInfo evolveIntoCard = cards.CardByName(evolveInto);
+
+                if (target != null && evolveIntoCard == null && target.IsOldApiCard()) // Maybe this is due to poor naming conventions allowed by the old API.
+                    evolveIntoCard = cards.CardByName($"{target.GetModPrefix()}_{evolveInto}");
+
+                if (target != null && evolveIntoCard != null)
+                    target.SetEvolve(evolveIntoCard, numberOfTurns, mods);
+
+                return cards;
+            };
+        }
+        else
+        {
+            info.SetEvolve(evolution, numberOfTurns, mods);
+        }
+        return info;
+    }
+
+    /// <summary>
+    /// Sets the ice cube parameters of the card. These parameters are used to make the IceCube ability function correctly.
+    /// </summary>
+    /// <param name="info">Card to access</param>
+    /// <param name="iceCube">The card that will be generated when this card dies.</param>
+    /// <param name="mods">A set of card mods to be applied to the ice cube contents</param>
+    /// <returns>The same card info so a chain can continue</returns>
+    public static CardInfo SetIceCube(this CardInfo info, CardInfo iceCube, IEnumerable<CardModificationInfo> mods = null)
+    {
+        info.iceCubeParams = new()
+        {
+            creatureWithin = iceCube
+        };
+
+        if (mods != null)
```

Comment:
This is a pending issue I haven't gotten around to fixing yet, but now is a good time: This should also check that mods.Count > 0